spec/rails_helper.rb

```diff
@@ -0,0 +1,102 @@
+# option on the command line or in ~/.rspec, .rspec or `.rspec-local`.
+#
+# The following line is provided for convenience purposes. It has the downside
+# of increasing the boot-up time by auto-requiring all files in the support
+# directory. Alternatively, in the individual `*_spec.rb` files, manually
+# require only the support files necessary.
+#
+# Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
+
+# Checks for pending migration and applies them before tests are run.
+# If you are not using ActiveRecord, you can remove this line.
+ActiveRecord::Migration.maintain_test_schema!
+
+RSpec.configure do |config|
+  include Warden::Test::Helpers
+  Warden.test_mode!
+  # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
+  config.fixture_path = "#{::Rails.root}/spec/fixtures"
+
+  # If you're not using ActiveRecord, or you'd prefer not to run each of your
+  # examples within a transaction, remove the following line or assign false
+  # instead of true.
+  config.use_transactional_fixtures = true
+
+  # RSpec Rails can automatically mix in different behaviours to your tests
+  # based on their file location, for example enabling you to call `get` and
+  # `post` in specs under `spec/controllers`.
+  #
+  # You can disable this behaviour by removing the line below, and instead
+  # explicitly tag your specs with their type, e.g.:
+  #
+  #     RSpec.describe UsersController, :type => :controller do
+  #       # ...
+  #     end
+  #
+  # The different available types are documented in the features, such as in
+  # https://relishapp.com/rspec/rspec-rails/docs
+  config.infer_spec_type_from_file_location!
+
+  # Filter lines from Rails gems in backtraces.
+  config.filter_rails_from_backtrace!
+  # arbitrary gems may also be filtered via:
+  # config.filter_gems_from_backtrace("gem name")
+  config.use_transactional_fixtures = false
+
+ config.before(:suite) do
+   if config.use_transactional_fixtures?
+     raise(<<-MSG)
+       Delete line `config.use_transactional_fixtures = true` from rails_helper.rb
+       (or set it to false) to prevent uncommitted transactions being used in
+       JavaScript-dependent specs.
+
+       During testing, the app-under-test that the browser driver connects to
+       uses a different database connection to the database connection used by
+       the spec. The app's database connection would not be able to access
+       uncommitted transaction data setup over the spec's database connection.
+     MSG
+   end
+   DatabaseCleaner.clean_with(:truncation)
+ end
+
+ config.before(:each) do
+   DatabaseCleaner.strategy = :transaction
+ end
+
+ config.before(:each, type: :feature) do
+   # :rack_test driver's Rack app under test shares database connection
+   # with the specs, so continue to use transaction strategy for speed.
+   driver_shares_db_connection_with_specs = Capybara.current_driver == :rack_test
```

Comment:
Metrics/LineLength: Line is too long. [81/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)